arcgis-ios-sdk-samples/Scenes/Orbit camera around object/OrbitCameraAroundObjectViewController.swift

```diff
@@ -0,0 +1,190 @@
+        
+        // Position the plane a third from the bottom of the screen.
+        cameraController.targetVerticalScreenFactor = 0.33
+        
+        // Don't pitch the camera when the plane pitches.
+        cameraController.isAutoPitchEnabled = false
+        return cameraController
+    }
+    
+    // MARK: Actions
+    
+    @IBAction func changeViewBarButtonItemTapped(_ sender: UIBarButtonItem) {
+        moveCameraAnimationCancelable?.cancel()
+        if sender.title == "Cockpit View" {
+            cockpitViewBarButtonItemTapped(sender)
+            sender.title = "Center View"
+        } else if sender.title == "Center View" {
+            centerViewBarButtonItemTapped(sender)
+            sender.title = "Cockpit View"
+        }
+    }
+    
+    func centerViewBarButtonItemTapped(_ sender: UIBarButtonItem) {
+        let cameraController = sceneView.cameraController as! AGSOrbitGeoElementCameraController
+        
+        cameraController.isCameraDistanceInteractive = true
+        cameraController.isAutoPitchEnabled = false
+        
+        cameraController.targetOffsetX = 0
+        cameraController.targetOffsetY = 0
+        cameraController.targetOffsetZ = 0
+        
+        cameraController.cameraHeadingOffset = 0
+        
+        cameraController.minCameraPitchOffset = 10
+        cameraController.maxCameraPitchOffset = 100
+        cameraController.cameraPitchOffset = 45
+        
+        cameraController.minCameraDistance = 10
+        cameraController.cameraDistance = 50
+    }
+    
+    func cockpitViewBarButtonItemTapped(_ sender: UIBarButtonItem) {
+        let cameraController = sceneView.cameraController as! AGSOrbitGeoElementCameraController
+        
+        cameraController.isCameraDistanceInteractive = false
+        cameraController.minCameraDistance = 0.1
+        // Unlock the camera pitch for the rotation animation.
+        cameraController.minCameraPitchOffset = -180
+        cameraController.maxCameraPitchOffset = 180
+        
+        // Animate the camera target to the cockpit.
+        cameraController.setTargetOffsetX(0, targetOffsetY: -2, targetOffsetZ: 1.1, duration: 1)
+        
+        // If the camera is already tracking the plane's pitch, subtract it from
+        // the delta angle for the animation.
+        let pitchDelta = cameraController.isAutoPitchEnabled ? 0 : 90 - cameraController.cameraPitchOffset + (planeGraphic.attributes["PITCH"] as! Double)
+        // Animate the camera so that it is at the target (cockpit), facing
+        // forward (0 deg heading), and aligned with the horizon (90 deg pitch).
+        moveCameraAnimationCancelable = cameraController.moveCamera(
+            withDistanceDelta: 0.1 - cameraController.cameraDistance,
+            headingDelta: -cameraController.cameraHeadingOffset,
+            pitchDelta: pitchDelta,
+            duration: 1
+        ) { [weak self] animationFinished in
+            self?.moveCameraAnimationCancelable = nil
```

Comment:
I'm following the discussion here: https://github.com/Esri/arcgis-runtime-samples-ios/pull/1074#discussion_r641710722 - nullify the cancelable task once it's finished.

The `cancel()` is called in `changeViewBarButtonItemTapped`, so that if the cancelable is non-nil (which means animation in action), we can cancel it before the camera moves.

Reply:
Ah, I see. I think I was thinking that this was explicitly calling `cancel()`, not just setting the cancelable to `nil`. Never mind!